Formula/wasm-micro-runtime.rb

```diff
@@ -0,0 +1,42 @@
+class WasmMicroRuntime < Formula
+  desc "WebAssembly Micro Runtime (WAMR)"
+  homepage "https://github.com/bytecodealliance/wasm-micro-runtime"
+  url "https://github.com/bytecodealliance/wasm-micro-runtime/archive/refs/tags/WAMR-05-18-2022.tar.gz"
+  version "2022.05.18"
+  sha256 "c71f571f99c7028fd7595126ec25d6df4bc9ba62702bb88cbd5ea8a29e477f12"
+  license "Apache-2.0"
+  head "https://github.com/bytecodealliance/wasm-micro-runtime.git", branch: "main"
+
+  depends_on "cmake" => :build
+
+  def install
+    cmake_args = std_cmake_args + %w[
+      -DWAMR_BUILD_MULTI_MODULE=1
+      -DWAMR_BUILD_DUMP_CALL_STACK=1
+      -DWAMR_BUILD_JIT=0
+      -DWAMR_BUILD_LIBC_UVWASI=0
+    ]
+
+    cmake_source = "product-mini/platforms/#{OS.mac? ? "darwin" : "linux"}"
+
+    system "cmake", "-S", cmake_source, "-B", "build", *cmake_args
```

Comment:
```suggestion
    cmake_args = %w[
      -DWAMR_BUILD_MULTI_MODULE=1
      -DWAMR_BUILD_DUMP_CALL_STACK=1
      -DWAMR_BUILD_JIT=0
      -DWAMR_BUILD_LIBC_UVWASI=0
    ]
    cmake_source = buildpath/"product-mini/platforms"/OS.kernel_name.downcase

    system "cmake", "-S", cmake_source, "-B", "platform_build", *cmake_args, *std_cmake_args
```
Why do we need to configure two separate CMake builds?

Reply:
Somehow I couldn't get both the CLI and the shared library out of the root level build only this way. Probably I should ask upstream if there is a way, but I don't think so.

Reply:
Yes, please ask. Please also add a comment documenting which CMake build is which (i.e. CLI, shared library).